server_tests/test/clientTests.js

```diff
@@ -1269,79 +1292,56 @@ describe("RUST API TEST SUITE", function() {
-            var startTime = Date.now();
-            var timeSendAnother = 1000;
-            var stopTime = 5000;
-            var stopSent = false;
-            var stopErr;
-            //instantiate the first camera._bublCaptureVideo command
-            return testClient.bublCaptureVideo(sessionId, function(res) {
-                if(Date.now() - startTime >= stopTime && !stopSent) {
-                    stopSent = true;
-                    return testClient.bublStop(res.body.id)
-                    .then(function(res) {
-                        stopErr = Comparison.bublStopOutput(res);
-                    });
-                }
-            })
-            .then( function(res) {
-                captureVideoDone = true;
-            });
-            try {
-                require('assert')(!captureVideoDone, 'first _bublCaptureVideo should not return at this point');
-            } catch(e) {
-                done(e);
-                return;
-            }
-            //instantiate the camera._bublTimelapse command
-            setTimeout(function() {
-                return testClient.bublTimelapse(sessionId, function(res) {})
-                .then( shouldNotGetHere, function onError (err) {
-                    //make sure first timelapse finished before entering next test
-                    var checkTimelapseDone;
-                    checkTimelapseDone = setInterval(function() {
-                        if(captureVideoDone) {
-                            if(stopErr) {
-                                done(stopErr);
-                            } else {
-                                Comparison.cameraInExclusiveUseError(err);
-                            }
-                            clearInterval(checkTimelapseDone);
-                        }
-                    }, 300);
-                });
-            }, timeSendAnother);
-        });
+            this.timeout(timeoutValue * 2);
+            var expectedResults = {
+              results: {
+                  _fileUris: ['']
+              }
+            };
+            var commandId;
+            var deferred = Q.defer();
+
+            return Q.all([testClient.bublCaptureVideo(sessionId, function onUpdate (res) {
+                if(!commandId) {
+                    commandId = res.body.id;
+                    return testClient.bublTimelapse(sessionId)
+                    .then(
+                        () => assert.fail("Should have received cameraInExclusiveUseError"),
+                        (err) => Comparison.cameraInExclusiveUseError(err)
+                    )
+                    .then(() => testClient.bublStop(commandId))
+                    .then((res) => Comparison.bublStopOutput(res))
+                    .then(deferred.resolve, deferred.reject)
+                  }
+                }), deferred.promise])
+              .then((res) => Comparison.bublCaptureVideoOutput(res[0], expectedResults), shouldNotGetHere)
+          });
```

Comment:
I don't believe it is actually ignored, rather it overwrites the default of an array with 2 strings, and replaces it as an array with 1 which passes since we are stopping the stream after 1.